src/main/java/fr/jmmc/oiexplorer/core/gui/chart/RulerOverlay.java

```diff
@@ -0,0 +1,209 @@
+        } else {
+            dataArea = plotInfo.getDataArea();
+        }
+
+        if (dataArea != null) {
+            final ValueAxis xAxis = plot.getDomainAxis();
+            final ValueAxis yAxis = plot.getRangeAxis();
+
+            final RectangleEdge xAxisEdge = plot.getDomainAxisEdge();
+            final RectangleEdge yAxisEdge = plot.getRangeAxisEdge();
+
+            double XYvalues[] = {
+                xAxis.valueToJava2D(x, dataArea, xAxisEdge),
+                yAxis.valueToJava2D(y, dataArea, yAxisEdge)
+            };
+            return XYvalues;
+        }
+        return null;
+    }
+
+    /**
+     * return the distance between the two points of the measure
+     *
+     * @return
+     */
+    private double calculateMeasure() {
+        return Math.sqrt(Math.pow((end[0] - origin[0]), 2) + Math.pow((end[1] - origin[1]), 2));
+    }
+
+    private double calculateAngle() {
+        return Math.toDegrees(Math.atan((end[0] - origin[0]) / (end[1] - origin[1])));
+    }
+
+    @Override
+    public void paintOverlay(Graphics2D g2, ChartPanel chartPanel) {
+        g2.clip(chartPanel.getChartRenderingInfo().getPlotInfo().getDataArea());
+        g2.setColor(Color.green);
+        g2.setStroke(new BasicStroke(2));
+
+        if (clickCounter != 0) {
+            double originCoords[] = translateValuesToCoords(origin[0], origin[1]);
+            double endCoords[] = translateValuesToCoords(end[0], end[1]);
+            g2.drawRect((int)originCoords[0] - RECT_SIZE, (int)originCoords[1] - RECT_SIZE, RECT_SIZE * 2, RECT_SIZE * 2);
+            g2.drawRect((int)endCoords[0] - RECT_SIZE, (int)endCoords[1] - RECT_SIZE, RECT_SIZE * 2, RECT_SIZE * 2);
+            g2.drawLine((int)originCoords[0], (int)originCoords[1], (int)endCoords[0], (int)endCoords[1]);
+            
+            g2.setClip(chartPanel.getChartRenderingInfo().getChartArea());
+            g2.setColor(Color.black);
+            g2.drawString("Point 1: x=" + String.format("%.5f", origin[0]) + " y=" + String.format("%.5f", origin[1]), 100, chartPanel.getSize().height - 120);
+            g2.drawString("Point 2: x=" + String.format("%.5f", end[0]) + " y=" + String.format("%.5f", end[1]), 100, chartPanel.getSize().height - 105);
+            g2.drawString("Measure: " + String.format("%.5f", calculateMeasure()) + " mas", 100, chartPanel.getSize().height - 60);
+            g2.drawString("Angle: " + String.format("%.5f", calculateAngle()) + " °", 100, chartPanel.getSize().height - 45);
+        }
+    }
+
+    @Override
+    public boolean support(final int eventType) {
+        return true;
+    }
+
+    @Override
+    public void chartMouseMoved(final ChartMouseEvent chartMouseEvent) {
+        if (isRulerActivated) {
+            setP2(chartMouseEvent.getTrigger().getX(), chartMouseEvent.getTrigger().getY());
+        }
+    }
+
+    @Override
+    public void chartMouseClicked(final ChartMouseEvent chartMouseEvent) {
+        if (!isRulerActivated) {
```

Comment:
this logic could be simplified with only 1 ruleState flag (disabled, editing, done) ?